testsuite/test_serverclass.py

```diff
@@ -0,0 +1,267 @@
+
+TEST_SCHEMA = os.path.join(SCHEMA_MOF_DIR, 'test_schema1.mof')
+
+VERBOSE = True
+
+
+class BaseMethodsForTests(object):
+    """
+    Common methods for test of server class.  This includes methods to
+    build the DMTF schema and to build individual instances.
+    """
+
+    def build_schema_list(self, conn, namespace):
+        """
+        Build the schema qualifier and class objects in the repository.
+        This requires only that the leaf objects be defined in a mof
+        include file since the compiler finds the files for qualifiers
+        and dependent classes.
+        TODO: ks 3/18 Right now we must build a file in the schema directory
+        to do this because the compile_from_string does not support
+        include files, etc. It does not search for files. See issue 1138
+        """
+        install_dmtf_schema()
+        class_list = """
+            #pragma locale ("en_US")
+            #pragma include ("Interop/CIM_RegisteredProfile.mof")
+            #pragma include ("Interop/CIM_Namespace.mof")
+            #pragma include ("Interop/CIM_ObjectManager.mof")
+            """
+        # See issue 1138.
+        # conn.compile_mof_str(class_list, namespace=namespace,
+        #                      search_paths=SCHEMA_MOF_DIR)
+        with open(TEST_SCHEMA, "w") as schema_file:
+            schema_file.write(class_list)
+
+        conn.compile_mof_file(TEST_SCHEMA, namespace=namespace,
+                              search_paths=[SCHEMA_MOF_DIR])
+
+        if os.path.isfile(TEST_SCHEMA):
+            os.remove(TEST_SCHEMA)
+        return
+
+    def inst_from_class(self, conn, cls_name, namespace=None, PropertyList=None,
+                        PropertyValues=None, IncludeNullProperties=True,
+                        strict=False, IncludePath=True):
+        """
+        Build an instance from the input classname.
+
+        Parameters:
+          conn (:class:`~pywbem.WBEMConnection`):
+            The WBEMConnection used to get the class from which
+            properties are derived for the instance.
+
+          cls_name
+            String with the class name from which properties are to
+            be derived for the instance.
+
+          namespace (:term:`string`):
+            Namespace in the WBEMConnection used to retrieve the class or
+            `None` if the default_namespace is to be used.
+
+          PropertyList (string or iterable of string)
+            Limits the properties retrieved witht the class and therefore
+            the properties in the resulting instance.
+
+          PropertyValues (dictionary):
+            Dictionary containing nane/value pairs where the names are the
+            names of properties in the class and the properties are the
+            property values to be set into the instance.
+
+          IncludeNullProperties (:class:`py:bool`:):
+            Determines if properties with Null values are included in the
+            instance.
```

Comment:
The description leaves it open as to whether this condition is applied as an additional restriction to the set of properties in PropertyList, vs. adding to them.

Reply:
That is now clear I think since I separated getting the class so that the instance is created from the class in inst_from_class which does not have the PropertyList parameter.